_posts/2025-04-28-sparse-autodiff.md

```diff
@@ -502,24 +514,49 @@ in this case addition, multiplication and the sign function.
 </div>
 
 As discussed in the previous section,
-all inputs are seeded with their respective input index sets.
-Figure 14 annotates these index sets on the edges of the computational graph.
-Our system for sparsity detection must now perform an **alternative evaluation of our computational graph**.
-Instead of computing the original function, 
-each operator must correctly propagate and accumulate the index sets of its inputs, 
-depending on whether an operator has a non-zero derivative or not.  
-
-Since addition and multiplication globally have non-zero derivatives with respect to both of their inputs, 
+all inputs are initialized with their respective singleton index sets.
+The sparsity detection system must walk through the computational graph in topological order.
+Each individual operator in the graph is "overloaded" to behave in a different way during sparsity detection.
+Instead of computing the output value, 
+it must correctly propagate and accumulate the index sets of its inputs, 
+depending on whether it has a non-zero derivative or not.  
+
+Figure 14 annotates index sets on the edges of the computational graph.
+Since addition and multiplication have non-zero derivatives with respect to both of their inputs, 
```

Comment:
Please revert my description of the figure.
You don't even know what I'm talking about here since I haven't added the figure yet... 😅

Reply:
See [this comment](https://github.com/adrhill/iclr2025-sparse-autodiff/pull/14/files#r1845252142), my main issue is about terminology